static/app/actionCreators/navigation.tsx

```diff
@@ -33,7 +33,7 @@ export function navigateTo(
           }
           onFinish={path => {
             modalProps.closeModal();
-            setTimeout(() => router.push(path), 0);
+            return window.setTimeout(() => router.push(path), 0);
```

Comment:
This is good to 🚢  👍  so you can ignore below for the purpose of this PR.

Thinking longer term here, but we probably want to fix this happening in the future as well, maybe doing something like changing all these `onFinish` to `NodeJS.Timeout | null` for these `onFinish` since they are likely to have a setTimeout it seems, meaning if someone adds a new one it will demand they add a return and might spark them to add the timeout?

You can even go further and use a nominal type (hack) to force everyone to explicitly do something like `return 0 as NoTimeout` but I'm not sure it's worth it for what is a fairly unlikely use case and the cognitive overhead of adding nominal types.

You can ignore all this as I've mentioned, just thinking aloud about how we might fix this "timeouts not cleared" in components. The only other thing I can think of is to do change all of our timeouts to our own timeout functions and assert that `setTimeout` generates an id that `clearTimeout` always gets called with.

Reply:
yes, I find this just confusing tbh. I like the idea of enforcing the timeout return. I'll add a comment on why we are returning a timeout here. I think we should think of how we can get rid of this logic where we call a function that renders a component.

Reply:
Yeah 👍